cmd/antrea-agent/agent.go

```diff
@@ -319,6 +321,13 @@ func run(o *Options) error {
 	if err != nil {
 		return fmt.Errorf("error when creating agent API server: %v", err)
 	}
+	check := healthz.NamedCheck("watcher", func(_ *http.Request) error {
+		if networkPolicyController.GetControllerConnectionStatus() {
+			return nil
+		}
+		return fmt.Errorf("Some watchers may not be connected")
+	})
+	apiServer.GenericAPIServer.AddHealthChecks(check)
```

Comment:
HealthCheck will be added to livezChecks too, what the issue asked was watcher's connection state should be reflected in readiness probe.